Content.Client/UserInterface/Systems/Chat/ChatUIController.cs

```diff
@@ -757,6 +774,14 @@ private void OnChatMessage(MsgChatMessage message)
 
     public void ProcessChatMessage(ChatMessage msg, bool speechBubble = true)
     {
+        // color the name unless it's something like "the old man"
+        if ((msg.Channel == ChatChannel.Local || msg.Channel == ChatChannel.Whisper) && _cfg.GetCVar(CCVars.ChatEnableColorName))
+        {
+            var grammar = _ent.GetComponentOrNull<GrammarComponent>(_ent.GetEntity(msg.SenderEntity));
+            if (grammar != null && grammar.ProperNoun == true)
+                msg.WrappedMessage = SharedChatSystem.InjectTagInsideTag(msg, "Name", "color", GetNameColor(SharedChatSystem.GetStringInsideTag(msg, "Name")));
```

Comment:
you can massively simplify all of this logic if, instead of implementing the tag swapping, you simply just always had it set a color and then base which color is chosen off of the CVAR.

so when the CVAR is false, just make the color always white (or get the default text color from the control) and when the CVAR is true make it get a color based off the name.

Reply:
Since it was asked that the coloration happens clientside, you still need to inject the color tag into the message regardless right? 

The alternative would be to send the color tag in the message from the server, but then you still need to dig into the tag to replace the color based on the local cvar.

Reply:
And to clarify, the [Name] tag is not replaced. It's there to ensure names are always of the same length in the different text strings, as some of them have additional tags to affect SpeechBubble behavior.

Reply:
my impression was just that it would likely be easier to set the color to a default rather than changing the tags of the message but that might not be the case.